packages/sysutils/busybox/scripts/rpi-flash-firmware

```diff
@@ -29,45 +25,25 @@ fi
 if [ -f "${FLAG_FILE}" ]; then
   . ${FLAG_FILE}
 
-  # Prepare environment - minimal config.txt, no eeproms/HATs
+  # Prepare flashing environment
   if [ "${MODE}" = "init" ]; then
-    rm -f "${BACKUP_CONFIG}" "${BACKUP_CMDLINE}"
-
-    [ -f "${CONFIG}" ] && mv "${CONFIG}" "${BACKUP_CONFIG}"
-    [ -f "${CMDLINE}" ] && mv "${CMDLINE}" "${BACKUP_CMDLINE}"
-
-    # "Safe" defaults
-    echo "gpu_mem=256" >>"${CONFIG}"
-    echo "dtoverlay=vc4-fkms-v3d" >>"${CONFIG}"
-
-    # Disable HAT eeproms, just in case...
-    echo "force_eeprom_read=0" >>"${CONFIG}"
-
-    # Create "safe" cmdline.txt based on minimal kernel ars
-    cmdline=
-    for kparam in $(cat /proc/cmdline); do
-      case "${kparam}" in
-        boot=*|disk=*|ip=*|smsc95xx.macaddr=*|overlay)
-          cmdline="${cmdline} ${kparam}";;
-      esac
-    done
-    echo "${cmdline:1} quiet" >"${CMDLINE}"
+    # Copy new SPI bootloader files, to be applied during next boot
+    if [ "${BOOTLOADER}" = "yes" ]; then
+      USE_FLASHROM=0 /usr/bin/rpi-eeprom-update -a
+    fi
 
     # Bump process to next step
     sed -e 's/^MODE=.*/MODE="update"/' -i "${FLAG_FILE}"
+    sync
   else
     rm -f "${FLAG_FILE}"
-
-    # Restore original config and cmdline if we have a backup
-    [ -f "${BACKUP_CONFIG}" ] && mv "${BACKUP_CONFIG}" "${CONFIG}"
-    [ -f "${BACKUP_CMDLINE}" ] && mv "${BACKUP_CMDLINE}" "${CMDLINE}"
-
     sync
 
     if [ "${MODE}" = "update" ]; then
-      # Apply SPI bootloader update
+      # Cleanup SPI bootloader files and show current version
       if [ "${BOOTLOADER}" = "yes" ]; then
-        SKIP_SAFE_MODE_RECOVERY=0 /usr/bin/rpi-eeprom-update -a
+        rm -f /flash/RECOVERY.* /flash/pieeprom.upd
```

Comment:
This is probably o.k on LibreElec but NOOBs has recovery.elf, recovery.rfs and recovery.cmdline

Reply:
The above should be OK in LibreELEC as `rm -f /flash/RECOVERY.*` will be case-sensitive, so it shouldn't conflict with anything that NOOBS might possibly have created (not entirely sure if NOOBS even creates these files in the LibreELEC `/flash` partition?)

But I can change this to `rm -f /flash/RECOVERY.[0-9][0-9][0-9]` to be on the safe side.